src/model/dump.ml

```diff
@@ -0,0 +1,271 @@
+module Identifier =struct
```

Comment:
This module is very similar to src/test/print/print.ml - if you've got anything extra can you merge it into there please.